spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -0,0 +1,159 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.actions;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.iceberg.StatisticsFile;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.actions.AnalyzeTable;
+import org.apache.iceberg.actions.ImmutableAnalyzeTable;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.puffin.StandardBlobTypes;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.spark.JobGroupInfo;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Computes the statistic of the given columns and stores it as Puffin files. */
+public class AnalyzeTableSparkAction extends BaseSparkAction<AnalyzeTableSparkAction>
+    implements AnalyzeTable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnalyzeTableSparkAction.class);
+
+  private final Table table;
+  private Set<String> columns = ImmutableSet.of();
+  private Set<String> types = StandardBlobTypes.blobTypes();
+  private Long snapshotId;
+
+  AnalyzeTableSparkAction(SparkSession spark, Table table) {
+    super(spark);
+    this.table = table;
+  }
+
+  @Override
+  protected AnalyzeTableSparkAction self() {
+    return this;
+  }
+
+  @Override
+  public Result execute() {
+    if (snapshotId == null) {
+      snapshotId = table.currentSnapshot().snapshotId();
+    }
+    String desc = String.format("Analyzing table %s for snapshot id %s", table.name(), snapshotId);
+    JobGroupInfo info = newJobGroupInfo("ANALYZE-TABLE", desc);
+    return withJobGroupInfo(info, this::doExecute);
+  }
+
+  private Result doExecute() {
+    LOG.info("Starting the analysis of {} for snapshot {}", table.name(), snapshotId);
+    List<AnalysisResult> analysisResults =
+        types.stream()
+            .map(
+                statsName -> {
+                  switch (statsName) {
+                    case StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1:
+                      return generateNDVAndCommit();
```

Comment:
in the future there will be multiple blob/stat types. the commit should be separate from blob generation